tiny_cnn/io/tensorflow/proto_parser.h

```diff
@@ -0,0 +1,256 @@
+          if (i > 0) ret += ", ";
+          for (const auto& d : attr_value.list().shape(i).dim()){
+            if (d.size() == -1)
+              ret += "   <What? Size is undefined here>";
+            else {
+              ret += ' ';
+              ret += d.size();
+            }
+          }
+        }
+      } else if (attr_value.list().tensor_size() > 0) {
+        for (int i = 0; i < attr_value.list().tensor_size(); ++i) {
+          if (i > 0) ret += ", ";
+          ret += attr_value.list().tensor(i).ShortDebugString();
+        }
+      }
+
+      ret += "]";
+      cout << ret << endl;
+      break;
+    }
+    case AttrValue::kFunc: {
+      string tmp_tostr;
+      for (auto p : attr_value.func().attr()) {
+        cout << "   SubAttr: " << p.first << ": " << endl;
+        summarize_attr_value(p.first, p.second);
+      }
+      break;
+    }
+    case AttrValue::kPlaceholder: {
+      string tmp_tostr;
+      all2str(tmp_tostr, attr_value.placeholder());
+      cout << "   (kPlaceholder) " << attr_string << ": " << tmp_tostr << endl;
+    }
+      break;
+    case AttrValue::VALUE_NOT_SET: {
+      cout << "   <Unknown AttrValue type> " << endl;
+    }
+      break;
+  }
+}
+
+bool IsPlaceholder(const tensorflow::NodeDef& node_def) {
+  if (node_def.op() != "Placeholder" || node_def.name() != "feed") {
+    return false;
+  }
+  bool found_dtype = false;
+  bool found_shape = false;
+  for (const auto& attr : node_def.attr()) {
+    if (attr.first == "dtype") {
+      if (attr.second.type() == tensorflow::DT_INT32) {
+        found_dtype = true;
+      } else {
+        return false;
+      }
+    } else if (attr.first == "shape") {
+      found_shape = true;
+    }
+  }
+  return found_dtype && found_shape;
+}
+
+bool IsScalarConst(const tensorflow::NodeDef& node_def) {
+  if (node_def.op() != "Const" || node_def.name() != "scalar") {
+    return false;
+  }
+  bool found_dtype = false;
+  bool found_value = false;
+  for (const auto& attr : node_def.attr()) {
+    if (attr.first == "dtype") {
+      if (attr.second.type() == tensorflow::DT_INT32) {
+        found_dtype = true;
+      } else {
+        return false;
+      }
+    } else if (attr.first == "value") {
+      if (attr.second.has_tensor() &&
+          attr.second.tensor().int_val_size() > 0) {
+        found_value = true;
+      } else {
+        return false;
+      }
+    }
+  }
+  return found_dtype && found_value;
+}
+
+// Iterates though all nodes in the GraphDef and prints info about them.
+void ListNodes(const tensorflow::GraphDef& graph_def) {
+  cout << "  Node Size: " << graph_def.node_size() << endl;
+  for (int i = 0; i < graph_def.node_size(); i++) {
+    const tensorflow::NodeDef& node_def = graph_def.node(i);
+    // print nodes information
+    fprintf(stderr, "\nNode: %s (%s)\n", node_def.name().c_str(), node_def.op().c_str());
+    fprintf(stderr, "  Is Placeholder: %d, Is Const: %d\n", IsPlaceholder(node_def), IsScalarConst(node_def));
```

Comment:
@Wangyida 
Is there any reason to use ```stderr```? If tiny-dnn use ```stderr``` even when we don't have any problem, it would confuse 
 users very much.